video/out/drm_common.c

```diff
@@ -497,19 +499,83 @@ static bool setup_mode(struct kms *kms, const char *mode_spec)
     return false;
 }
 
-static int open_card(int card_no)
+static int open_card_path(char *path)
 {
-    char card_path[128];
-    snprintf(card_path, sizeof(card_path), DRM_DEV_NAME, DRM_DIR_NAME, card_no);
-    return open(card_path, O_RDWR | O_CLOEXEC);
+    return open(path, O_RDWR | O_CLOEXEC);
+}
+
+static char *get_primary_device_path(struct mp_log *log,
+                                     int *given_card_no)
+{
+    drmDevice *devices[DRM_MAX_MINOR] = { 0 };
+    int card_count = drmGetDevices(devices,
+                                   sizeof(devices) / sizeof(devices[0]));
+    char *device_path = NULL;
+    bool card_no_given = (*given_card_no >= 0);
+
+    if (card_count < 0) {
+        mp_err(log, "Listing DRM devices with drmGetDevices failed! (%s)\n",
+               mp_strerror(errno));
+    }
+
+    if (card_no_given && *given_card_no > (card_count - 1)) {
+        mp_err(log, "Card number %d given too high! %d devices located.\n",
+               *given_card_no, card_count);
+        goto err;
+    }
+
+    for (int card_no = card_no_given ? *given_card_no : 0;
```

Comment:
I think the following code would be a bit more readable if the loop variable was just `i` (shorter)

Reply:
Done.